ilastik/experimental/api/_pipelines.py

```diff
@@ -0,0 +1,256 @@
+import warnings
+from functools import singledispatch
+
+import numpy
+import vigra
+import xarray
+
+from ilastik.applets.featureSelection.opFeatureSelection import OpFeatureSelection
+from ilastik.applets.objectClassification.opObjectClassification import OpMultiRelabelSegmentation, OpObjectPredict
+from ilastik.applets.objectExtraction.opObjectExtraction import OpObjectExtraction
+from ilastik.applets.thresholdTwoLevels.opThresholdTwoLevels import OpThresholdTwoLevels
+from ilastik.experimental import parser
+from lazyflow.graph import Graph
+from lazyflow.operators import OpMultiArrayStacker, OpReorderAxes, OpPixelOperator
+from lazyflow.operators.classifierOperators import OpClassifierPredict
+
+from .types import (
+    ObjectClassificationFromSegmentationPipeline,
+    ObjectClassificationFromPredictionPipeline,
+    PixelClassificationPipeline,
+)
+
+
+def _reorder_output(output: numpy.ndarray, output_axisorder: vigra.AxisTags, input_axisorder: str):
+    assert len(output.shape) == len(output_axisorder)
+    if "c" in output_axisorder and "c" not in input_axisorder:
+        # if input data was supplied without channel axes, put channel last
+        input_axisorder = input_axisorder + "c"
+
+    drop_axes = []
+    for axis, size in zip(output_axisorder, output.shape):
+        if axis not in input_axisorder:
+            assert size == 1
+            drop_axes.append(axis)
+
+    output_array = xarray.DataArray(output, dims=tuple(output_axisorder))
+    return output_array.squeeze(drop_axes).transpose(*tuple(input_axisorder))
+
+
+@singledispatch
+def convert_to_vigra(data):
+    raise NotImplementedError(f"{type(data)}")
+
+
+@convert_to_vigra.register
+def _(data: vigra.VigraArray):
+    return data
+
+
+@convert_to_vigra.register
+def _(data: numpy.ndarray):
+    raise ValueError("numpy arrays don't provide information about axistags expecting xarray")
+
+
+@convert_to_vigra.register
+def _(data: xarray.DataArray):
+    axistags = "".join(data.dims)
+    return vigra.taggedView(data.values, axistags)
+
+
+class _PixelClassificationPipelineImpl(PixelClassificationPipeline):
```

Comment:
I'm not sure that we really need an ABC and it's implementation, just for the sake of a single method...

Do we really need to keep all operator state between calls? If no, then all prediction logic could be placed into project classes themselves since they already have workflow-specific types. Those methods would prepare all the required operators, run the computation and return the result, discarding all operator state.